### DISCUSSION TITLE
A question about LATCH

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#1007] [Questions]
When I use `LATCH`, I can't understand what 24-28 bits mean. For example, if I set a region bit to 3, will the 24-28 bits store a numeric value of 11000? If I use `beamdp` to read out particles interacting in region 3, can I select 24 and 25 in inclusive bits? There is another question, I am not very clear about the definition of secondary particles, what interactions occur to produce secondary particles? Are the particles after elastic collisions（Rayleigh scattering）secondary?

**Comment by rtownson** [comment]
If you read 24-28 as a byte, it gives you an integer number. That integer corresponds to the region number where the particle was created (all 0's, then it's primary). Secondaries are defined as any particle created that is not a primary from the source (some sources like a phsp may include secondaries). Rayleigh scattering does not produce any new particles, so the photon remains primary.

**Reply by CQUswl** [reply]
When I use beamdp to process the results, I find an interesting phenomenon, when I set bit1 in beamnrc, I select bit24 in inclusive bits and get the same result as I select bit24 plus any bit in bit25 to bit28.

**Reply by CQUswl** [reply]
> If you read 24-28 as a byte, it gives you an integer number. That integer corresponds to the region number where the particle was created (all 0's, then it's primary). Secondaries are defined as any particle created that is not a primary from the source (some sources like a phsp may include secondaries). Rayleigh scattering does not produce any new particles, so the photon remains primary.

Can I understand that when I only set region 1, the difference between bit1 and bit24 is that bit1 contains interactions such as Rayleigh scattering that do not produce secondary particles.

**Reply by rtownson** [reply]
Using only bit 24 is meaningless. You need to convert bits 24-28 together into an integer.

The bit1 is set to 1 if a particle interacted or passed through the region that you have assigned to flag bit1. The interacting vs. passing option is controlled by the LATCH_OPTION parameter. Rayleigh scattering counts as an interaction thus will set bit1 to 1.

**Reply by CQUswl** [reply]
> Using only bit 24 is meaningless. You need to convert bits 24-28 together into an integer.
> 
> The bit1 is set to 1 if a particle interacted or passed through the region that you have assigned to flag bit1. The interacting vs. passing option is controlled by the LATCH_OPTION parameter. Rayleigh scattering counts as an interaction thus will set bit1 to 1.

![1686370855497](https://github.com/nrc-cnrc/EGSnrc/assets/114200268/dcc02d31-59ea-46e7-8ca3-db12b3d3d267)
When I use beamdp to select bits, do I only need to select bit24 to represent selected area 1?Like the image above?

**Reply by rtownson** [reply]
I recommend ignoring bits 24-28 for now. It's easier to start by configuring your simulation to set one of the latch bits, e.g., bit 2. In the beamnrc_gui, when you define the material for any region, there is an option to set the latch bits for that region. Set it to 2 for your region of interest, and make sure that only your region of interest sets latch 2. Then, if the particle has interacted/passed through that region, bit 2 will be set to 1 (default is 0). Then you can use the "inclusive bits" filtering to view only particles that interacted/passed through that region, by checking box 2.
